frontend/src/lib/StatusUtils.ts

```diff
@@ -116,3 +129,76 @@ function wasNodeCached(node: NodeStatus): boolean {
     ? false
     : artifacts.some(artifact => artifact.s3 && !artifact.s3.key.includes(node.id));
 }
+
+// separate these helper function for paritial v2 api integration
+export function hasFinishedV2(state?: V2beta1RuntimeState): boolean {
+  switch (state) {
+    case V2beta1RuntimeState.SUCCEEDED: // Fall through
+    case V2beta1RuntimeState.SKIPPED: // Fall through
+    case V2beta1RuntimeState.FAILED: // Fall through
+    case V2beta1RuntimeState.CANCELED:
+      return true;
+    case V2beta1RuntimeState.PENDING: // Fall through
+    case V2beta1RuntimeState.RUNNING: // Fall through
+    case V2beta1RuntimeState.CANCELING: // Fall through
+    case V2beta1RuntimeState.RUNTIMESTATEUNSPECIFIED:
+      return false;
+    default:
+      return false;
+  }
+}
+
+export function statusToBgColorV2(state?: V2beta1RuntimeState, nodeMessage?: string): string {
+  state = checkIfTerminatedV2(state, nodeMessage);
+  switch (state) {
+    case V2beta1RuntimeState.FAILED:
+      return statusBgColors.error;
+    case V2beta1RuntimeState.PENDING:
+      return statusBgColors.notStarted;
+    case V2beta1RuntimeState.CANCELING:
+    // fall through
+    case V2beta1RuntimeState.RUNNING:
+      return statusBgColors.running;
+    case V2beta1RuntimeState.SUCCEEDED:
+      return statusBgColors.succeeded;
+    case V2beta1RuntimeState.SKIPPED:
+    // fall through
+    case V2beta1RuntimeState.CANCELED:
+      return statusBgColors.terminatedOrSkipped;
+    case V2beta1RuntimeState.RUNTIMESTATEUNSPECIFIED:
+    // fall through
+    default:
+      logger.verbose('Unknown state:', state);
+      return statusBgColors.notStarted;
+  }
+}
+
+export function checkIfTerminatedV2(
+  state?: V2beta1RuntimeState,
+  nodeMessage?: string,
+): V2beta1RuntimeState | undefined {
+  // Argo considers terminated runs as having "Failed", so we have to examine the failure message to
+  // determine why the run failed.
+  if (state === V2beta1RuntimeState.FAILED && nodeMessage === 'terminated') {
+    state = V2beta1RuntimeState.CANCELED;
+  }
+  return state;
+}
+
+export function parseNodePhaseV2(node: NodeStatus): NodePhase {
```

Comment:
Where is this function being used? Are the logic here and below functions still applicable to V2 scenario?

Reply:
Removed it.